### ISSUE TITLE
generatePackageJson created with no dependencies in nestjs project after 13->14 upgrade

### ISSUE THREAD

**Issue by Nick-Lucas** [nrwl/nx#10884]
This project was originally created on nx 13 and we are using the generatePackageJson option for our nestjs api to ship minimal dependencies.

## Current Behavior

After an otherwise successful upgrade to nx 14, the generated package json is now empty:

```json
{
  "name": "api",
  "version": "0.0.1",
  "dependencies": {},
  "main": "main.js"
}
```

## Expected Behavior

The generated package.json contains all dependencies which the nestjs api consumes. This was already working on NX 13

## Steps to Reproduce

In a fresh workspace I cannot reproduce this. It seems likely there's some under-the-hood error which is not being included in the console output

### Environment

```
   Node : 14.19.0
   OS   : linux x64
   yarn : 1.22.18
   
   nx : 14.3.6
   @nrwl/angular : Not Found
   @nrwl/cypress : 14.3.6
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.3.6
   @nrwl/eslint-plugin-nx : 14.3.6
   @nrwl/express : Not Found
   @nrwl/jest : 14.3.6
   @nrwl/js : 14.3.6
   @nrwl/linter : 14.3.6
   @nrwl/nest : 14.3.6
   @nrwl/next : Not Found
   @nrwl/node : 14.3.6
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 14.3.6
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 14.3.6
   @nrwl/web : 14.3.6
   @nrwl/workspace : 14.3.6
   typescript : 4.7.2
   ---------------------------------------
   Community plugins:
```

The project.json for the API looks like this:

```json
{
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/api/src",
  "projectType": "application",
  "targets": {
    "build": {
      "executor": "@nrwl/node:webpack",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/api",
        "main": "packages/api/src/main.ts",
        "tsConfig": "packages/api/tsconfig.app.json",
        "assets": ["packages/api/src/assets"]
      },
      "configurations": {
        "production": {
          "optimization": false,
          "inspect": false,
          "generatePackageJson": true,
          "fileReplacements": [
            {
              "replace": "packages/api/src/main.ts",
              "with": "packages/api/src/main.lambda.ts"
            },
            {
              "replace": "packages/api/src/environments/environment.ts",
              "with": "packages/api/src/environments/environment.prod.ts"
            }
          ]
        }
      }
    },
    "serve": {
      "executor": "@nrwl/node:node",
      "options": {
        "buildTarget": "api:build"
      },
      "configurations": {
        "production": {
          "buildTarget": "api:build:production"
        }
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/api/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/packages/api"],
      "options": {
        "jestConfig": "packages/api/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}

```

**Comment by Nick-Lucas** [comment]
It looks like the project graph being passed to [workspace/src/utilities/create-package-json](https://github.com/nrwl/nx/blob/master/packages/workspace/src/utilities/create-package-json.ts) is not complete. Essentially it's missing data in the graph.dependencies object.

Minimal workspace:

```
{
  "version": "5.0",
  "nodes": {
    ...omitted for brevity
  },
  "externalNodes": {
    ...omitted for brevity
  },
  "dependencies": {
    "nxrepro": [
      {
        "source": "nxrepro",
        "target": "npm:@nestjs/testing",
        "type": "static"
      },
      {
        "source": "nxrepro",
        "target": "npm:@nestjs/common",
        "type": "static"
      },
      {
        "source": "nxrepro",
        "target": "npm:@nestjs/core",
        "type": "static"
      }
    ]
  }
}
```

Our project workspace. All dependencies are missing any content in their dependency lists:
```json
{
  "version": "5.0",
  "nodes": {
    ...omitted for brevity
  },
  "externalNodes": {
    ...omitted for brevity
  },
  "dependencies": {
    "okta-management-types": [],
    "appdynamics-browser": [],
    "appdynamics-node": [],
    "okta-management": [],
    "common-browser": [],
    "okta-browser": [],
    "rest-browser": [],
    "rest-queries": [],
    "db-datajob": [],
    "portal-e2e": [
      {
        "source": "portal-e2e",
        "target": "portal",
        "type": "implicit"
      }
    ],
    "admin-e2e": [
      {
        "source": "admin-e2e",
        "target": "admin",
        "type": "implicit"
      }
    ],
    "api-types": [],
    "db-types": [],
    "db-dev": [],
    "portal": [],
    "admin": [],
    "theme": [],
    "utils": [],
    "api": []
  }
}
```

**Comment by Nick-Lucas** [comment]
Okay so looks like the jsPluginConfig passed into https://github.com/nrwl/nx/blob/c3b7db8c0ec6c0221a06907beae5cf30e03d9474/packages/nx/src/project-graph/build-dependencies/build-explicit-typescript-and-package-json-dependencies.ts is to blame. 

By default it has `{ analyzeSourceFiles: false }` and modifying the code to run this branch regardless causes Nx to output dependencies to the package.json correctly

**Comment by Nick-Lucas** [comment]
Now that I've been through all the debugging and found the root cause, the search function was finally able to find a solved issue, this is a duplicate of #10785 so I will close this
